presentation-api/controlling-ua/idlharness.html

```diff
@@ -0,0 +1,102 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Presentation API IDL tests for Controlling User Agent</title>
+<link rel="author" title="Louay Bassbouss" href="http://www.fokus.fraunhofer.de">
+<link rel="help" href="http://w3c.github.io/presentation-api/#dfn-controlling-user-agent">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/WebIDLParser.js"></script>
+<script src="/resources/idlharness.js"></script>
+
+<script id="untested_idl" type="text/plain">
+interface Navigator {
+};
+interface EventTarget {
+};
+interface EventHandler {
+};
+interface Event {
+};
+dictionary EventInit {
+};
+interface PresentationReceiver {
+};
+</script>
+
+<script id='idl' type="text/plain">
+partial interface Navigator {
+    [SameObject]
+    readonly attribute Presentation presentation;
+};
+
+interface Presentation {
+    attribute PresentationRequest? defaultRequest;
+    [SameObject]
+    readonly attribute PresentationReceiver? receiver;
+};
+
+[Constructor(DOMString url)]
+interface PresentationRequest : EventTarget {
+    Promise<PresentationConnection> start();
+    Promise<PresentationConnection> reconnect(DOMString presentationId);
+    Promise<PresentationAvailability> getAvailability();
+    attribute EventHandler onconnectionavailable;
+};
+
+interface PresentationAvailability : EventTarget {
+    readonly attribute boolean  value;
+    attribute EventHandler onchange;
+};
+
+[Constructor(DOMString type, PresentationConnectionAvailableEventInit eventInitDict)]
+interface PresentationConnectionAvailableEvent : Event {
+    [SameObject]
+    readonly attribute PresentationConnection connection;
+};
+
+dictionary PresentationConnectionAvailableEventInit : EventInit {
+    required PresentationConnection connection;
+};
+
+enum PresentationConnectionState {
+    "connected",
+    "closed",
+    "terminated"
+};
+
+enum BinaryType {
+    "blob",
+    "arraybuffer"
+};
+
+interface PresentationConnection : EventTarget {
+    readonly attribute DOMString? id;
+    readonly attribute PresentationConnectionState state;
+    void close();
+    void terminate();
+    attribute EventHandler onstatechange;
```

Comment:
Hmm, seems the spec has some changes at https://github.com/w3c/presentation-api/pull/227

Would you please update the IDL accordingly? Or shall we merge this as-is and address IDL update in following PRs?

Maybe you can add a comment in the test file to reflect the spec date for tracking.